File: java-serialization/src/main/java/com/codebreeze/serialization/json/model/Json.java
package com.codebreeze.serialization.json.model;

import org.apache.commons.lang3.builder.RecursiveToStringStyle;
import org.apache.commons.lang3.builder.ReflectionToStringBuilder;

import java.util.List;

public class Json {

    public static final RecursiveToStringStyle STYLE = new RecursiveToStringStyle();

    public static class AddressBook {
        private final String description;
        private final List<Person> person;

        public AddressBook(){
            description = null;
            person = null;
        }

        public AddressBook(final String description, final List<Person> person) {
            this.description = description;
            this.person = person;
        }

        @Override
        public String toString(){
            return new ReflectionToStringBuilder(this, STYLE).toString();
        }
    }

    public static class Person {
        private final String name;
        private final int id;
        private final String email;
        private final List<PhoneNumber> phoneNumber;

        public Person() {
            this.id = 0;
            this.name = null;
            this.email = null;
            this.phoneNumber = null;
        }

        public Person(final String name, final int id, final String email, final List<PhoneNumber> phoneNumber) {
            this.id = id;
            this.name = name;
            this.email = email;
            this.phoneNumber = phoneNumber;
        }
    }

    public static class PhoneNumber {
        private final String number;
        private final PhoneType phoneType;

        public PhoneNumber() {
            this.number = null;
            this.phoneType = null;
        }

        public PhoneNumber(final String number, final PhoneType phoneType) {
            this.number = number;
            this.phoneType = phoneType;
        }
    }

    public enum PhoneType {
        MOBILE,
        HOME,
        WORK;
    };
}

File: node-serialization/json.js
var random = require("random-js")(); // uses the nativeMath engine

function randomAddressBook() {
    return createAddressBook();
}

function createAddressBook() {
    var description = random.string(300);
    var person1 = createPerson();
    var person2 = createPerson();
    var person3 = createPerson();
    return {
        description: description,
        person: [person1, person2, person3]
    };
}

function toAddressBook(data) { //string
    return JSON.parse(data);
}

function toString(addressBook) {
    return JSON.stringify(addressBook);
}


function createPerson() {
    var phoneNumber1 = createPhoneNumber();
    var phoneNumber2 = createPhoneNumber();
    var name = random.string(25);
    var email = random.string(20);
    var id = random.integer(0, 10000000);
    return {
        name: name,
        id: id,
        email: email,
        phoneNumber: [phoneNumber1, phoneNumber2]
    }
}

function createPhoneNumber() {
    var phoneNumber = random.string(10, '0123456789');
    var phoneType = random.pick(['MOBILE', 'HOME', 'WORK']);
    return {
        phoneNumber: phoneNumber,
        phoneType: phoneType
    };
}

//var anAddressBook = randomAddressBook();
//console.log(JSON.stringify(anAddressBook, null, "  "));
//console.log(toString(anAddressBook));
//console.log(toAddressBook(toString(anAddressBook)));
